internal/extproc/translator/util.go

```diff
@@ -36,3 +43,56 @@ func parseDataURI(uri string) (string, []byte, error) {
 	}
 	return contentType, bin, nil
 }
+
+// buildGCPRequestMutations creates header and body mutations for GCP requests
+// It sets the ":path" header, the "content-length" header and the request body.
+func buildGCPRequestMutations(path string, reqBody []byte) (*ext_procv3.HeaderMutation, *ext_procv3.BodyMutation) {
```

Comment:
this mutation does not look like GCP specific, can we make it rename it? Also this needs a test

Reply:
added